src/google-drive/authentication.js

```diff
@@ -0,0 +1,81 @@
+let fs = require('fs');
+let readline = require('readline');
+let googleAuth = require('google-auth-library');
+
+let SCOPES = ['https://www.googleapis.com/auth/spreadsheets','https://www.googleapis.com/auth/drive.appdata','https://www.googleapis.com/auth/drive','https://www.googleapis.com/auth/drive.file']; //you can add more scopes according to your permission need. But in case you chang the scope, make sure you deleted the ~/.credentials/sheets.googleapis.com-nodejs-quickstart.json file
+const TOKEN_DIR = (process.env.HOME || process.env.HOMEPATH || process.env.USERPROFILE) + '/.credentials/'; //the directory where we're going to save the token
+const TOKEN_PATH = TOKEN_DIR + 'sheets.googleapis.com-nodejs-quickstart.json'; //the file which will contain the token
+
+class Authentication {
+  authenticate(){
+    return new Promise((resolve, reject)=>{
+      let credentials = this.getClientSecret();
+      let authorizePromise = this.authorize(credentials);
+      authorizePromise.then(resolve, reject);
+    });
+  }
+  getClientSecret(){
+    return require('./credentials.json');
+  }
+  authorize(credentials) {
+    var clientSecret = credentials.installed.client_secret;
+    var clientId = credentials.installed.client_id;
+    var redirectUrl = credentials.installed.redirect_uris[0];
+    var auth = new googleAuth();
+    var oauth2Client = new auth.OAuth2(clientId, clientSecret, redirectUrl);
+
+    return new Promise((resolve, reject)=>{
+      // Check if we have previously stored a token.
+      fs.readFile(TOKEN_PATH, (err, token) => {
```

Comment:
This should probably be stored in the database b/c this token will be per-installation